photon.h

```diff
@@ -48,17 +48,25 @@ const uint64_t INIT_IO_DEFAULT = INIT_IO_LIBCURL;
 
 #undef SHIFT
 
+struct PhotonOptions {
+    int libaio_ctx = 2048;
+    bool use_pooled_stack_allocator = false;
+    bool bypass_threadpool = false;
+};
+
 /**
  * @brief Initialize the main photon thread and ancillary threads by flags.
  *        Ancillary threads will be running in background.
  * @return 0 for success
  */
 int init(uint64_t event_engine = INIT_EVENT_DEFAULT,
-         uint64_t io_engine = INIT_IO_DEFAULT);
+         uint64_t io_engine = INIT_IO_DEFAULT,
+         const PhotonOptions& options = PhotonOptions());
```

Comment:
const PhotonOptions& options = {});

Reply:
without ```&``` may be more efficient for small structs

Reply:
The `PhotonOptions` might be much larger in future, maybe..?